util.py

```diff
@@ -0,0 +1,97 @@
+from urllib.parse import unquote
+
+from bs4 import BeautifulSoup
+import fetch
+
+
+def coupon_extract(start_position=0, end_position=0, html=""):
```

Comment:
This would probably be better done with regex (famous last words, I know). Something like
```python
match = re.search(r'"couponCode: (?P<code>.*?)"', element_text)
code = match.group("code")
```
Just a thought-  not a suggestion